ingestion-edge/ingestion_edge/publish.py

```diff
@@ -70,10 +82,15 @@
 
 def init_app(app: Sanic) -> Tuple[PublisherClient, SQLiteAckQueue]:
     """Initialize Sanic app with url rules."""
-    # Get PubSub timeout
-    timeout = app.config.get("PUBLISH_TIMEOUT_SECONDS")
     # Initialize PubSub client
+    timeout = app.config.get("PUBLISH_TIMEOUT_SECONDS", None)
     client = PublisherClient()
+    client.api.publish = partial(
+        client.api.publish,
+        retry=Retry(TRANSIENT_ERRORS, deadline=timeout),
+        timeout=timeout,
+    )
```

Comment:
python threads are unkillable, so they have to time out on their own. The publish api call happens in a thread, so we need to use the built in `retry` and `timeout` or we get unusual behavior where everyone waiting times out, but the api call is still retrying in a thread.

Reply:
this is also why `FLUSH_PUBLISH_TIMEOUT_SECONDS` went away, because we don't control batching we have to set a timeout to apply to all batches.